### PR TITLE
Remove pow and float_power TestGradient Skips

### PR DESCRIPTION
**Pull request by drisspg**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #76825

### CODE REVIEW

**Comment by drisspg** [comment]
Accidentally formatted the common_methods_invocation.py file while in a weird rebase thing going on. In the process of fixing that up I ended up making a new commit and ghstack made a new PR. NOOB problems.. but the history of this is way nicer and hopefully..... I see all green and this is ready to merge

@mruberry @soulitzer

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -13655,29 +13656,60 @@ def generate_std_var_kwargs(t: torch.Tensor, **kwargs):
                     supports_fwgrad_bwgrad=True,
                     assert_autodiffed=True,
                     supports_one_python_scalar=True,
-                    # TODO: FIXME: pow needs a way of specifying that for integer
-                    #   types only it does not support negative exponentes
-                    rhs_make_tensor_kwargs=dict(low=1),
+                    # Raising integer values to negative values is not causes an error
```
Comment (soulitzer) [inline torch/testing/_internal/common_methods_invocations.py:13659]: "is not causes an error"?

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -13655,29 +13656,60 @@ def generate_std_var_kwargs(t: torch.Tensor, **kwargs):
...
+                    rhs_make_tensor_kwargs=dict(low=0),
+                    # Logarithims of negative values are outside the domain of the function
```
Comment (soulitzer) [inline torch/testing/_internal/common_methods_invocations.py:13661]: nit: logarithim -> logarithm

What do you mean by this comment? Do you mean to say something about raising negative numbers to fractional powers?

File: torch/testing/_internal/common_methods_invocations.py (hunk 1 above)
Comment (soulitzer) [reply torch/testing/_internal/common_methods_invocations.py:13659]: Ohh nice also looks like low = 1 -> low = 0 now, that would make more sense

File: torch/testing/_internal/common_methods_invocations.py [hunk 3]
```diff
@@ -13655,29 +13656,60 @@ def generate_std_var_kwargs(t: torch.Tensor, **kwargs):
...
     BinaryUfuncInfo('float_power',
+                    ref=np.float_power,
                     dtypes=all_types_and_complex_and(torch.half, torch.bfloat16, torch.bool),
                     promotes_int_to_float=True,
                     supports_forward_ad=True,
                     supports_fwgrad_bwgrad=True,
                     supports_one_python_scalar=True,
+                    # Integer types do not support negative exponentes
+                    rhs_make_tensor_kwargs=dict(low=0),
+                    # Logarithims of negative values are outside the domain of the function
```
Comment (soulitzer) [inline torch/testing/_internal/common_methods_invocations.py:13693]: Same as above

File: torch/testing/_internal/common_methods_invocations.py [hunk 4]
```diff
@@ -13655,29 +13656,60 @@ def generate_std_var_kwargs(t: torch.Tensor, **kwargs):
...
                     skips=(
-                        # nan vs nan comparisons
-                        # https://github.com/pytorch/pytorch/issues/74279
-                        DecorateInfo(unittest.skip("Skipped!"), 'TestGradients'),
+                        # Skipping integers because they are being raised to negative powers causing an error
+                        DecorateInfo(unittest.expectedFailure, 'TestBinaryUfuncs', 'test_reference_numerics_small_values',
+                                     dtypes=[torch.int8, torch.int16, torch.int32, torch.int64]),
+                        DecorateInfo(unittest.expectedFailure, 'TestBinaryUfuncs', 'test_reference_numerics_large_values',
+                                     dtypes=[torch.int16, torch.int32, torch.int64]),
+                        # FIXME Complex values error with: Greatest absolute difference: nan at index
```
Comment (soulitzer) [inline torch/testing/_internal/common_methods_invocations.py:13676]: Did you want to file an issue for this?

**Review by soulitzer — APPROVED** [review]
Thanks! Just some small notes about the comments

File: torch/testing/_internal/common_methods_invocations.py (hunk 4 above)
Comment (drisspg) [reply torch/testing/_internal/common_methods_invocations.py:13676]: yup will create one now

Comment (drisspg) [reply torch/testing/_internal/common_methods_invocations.py:13676]: #76853

**Pull request closed without merging**